src/elfxx.c

```diff
@@ -154,9 +155,104 @@ elf_w (lookup_symbol) (unw_addr_space_t as UNUSED,
         }
       shdr = (Elf_W (Shdr) *) (((char *) shdr) + ehdr->e_shentsize);
     }
+
+  return ret;
+}
+
+static int
+elf_w (lookup_symbol_fn)(void *opaque, char *strtab,
+                         const Elf_W (Sym) *sym,
+                         Elf_W (Addr) real_start_ip)
+{
+  int ret = -UNW_ENOINFO;
+
+  struct {
+    unw_word_t ip;
+    char *buf;
+    size_t buf_len;
+    Elf_W (Addr) *min_dist;
+  } *arg = opaque;
```

Comment:
I'm not sure I understand why this is passed as an opaque pointer with the struct redefined in several places. Since these are all static functions would it not be better to declare the args struct once and pass it explicitly?